Unify website deployment logic with rpm-web

dmnks opened the pull request:
Drop duplicated layout files originally pulled from [rpm-web](https://github.com/rpm-software-management/rpm-web) and simplify `make site` while at it, we can now use plain `jekyll build` instead of going through podman since no plugins need to be installed.

pmatilai commented on .github/workflows/gh-pages.yml:
```diff
@@ -22,11 +22,16 @@ concurrency:
   # Build job
   build:
-    runs-on: ubuntu-24.04
+    runs-on: ubuntu-latest
```
I'd rather have us locked in on a known version, like we do with rpm. So rather than change this to -latest, change rpm-web to whatever the version we want just now.

I hate the kind of surprises where s*** breaks without notice. Maybe the risk of breakage isn't that bad with the web stuff as it is for rpm itself but still...

dmnks replied:
Ah good point, it's not like this never happened, quite the contrary :sweat_smile: Will fix (in rpm-web too).

pmatilai commented:
That version-issue aside, unifying stuff with rpm-web sounds like the sane thing to do :+1:

dmnks commented:
Fixed the runner version, and also added a forgotten short note to the last commit about the dropped `JEKYLL_*` options that weren't that useful to start with.

dmnks commented:
Thanks, and done: https://github.com/rpm-software-management/rpm-web/pull/63
